### ISSUE TITLE
Unclear message if named tuple is not compatible with protocol

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#7224]
Mypy just says that class is not compatible with the bound when type checking this example:

```py
from typing import Protocol, Generic, TypeVar, NamedTuple, List, Any

class Example(Protocol):
    sentence: List[str]

T = TypeVar('T', bound=Example)

class Dataset(Generic[T]):
    ...

class ConcreteExample(NamedTuple):
    sentence: List[str]
    others: Any

class ConcreteDataset(Dataset[ConcreteExample]):
    ...
```

Here's the error message:

```
t.py:15: error: Type argument "Tuple[builtins.list[builtins.str], Any, fallback=t.ConcreteExample]" of "Dataset" must be a subtype of "t.Example"
```

It would be better to also say that `sentence` in `ConcreteExample` is read-only, and maybe suggest making it a read-only property in the protocol.

Originally reported in #7212.

**Comment by jmehnle** [comment]
I'm having this problem. I have a user-defined protocol, and `mypy` is complaining that a concrete class is not a subtype of the protocol, but it gives me zero hints on how the class deviates from the protocol. :(

**Comment by gvanrossum** [comment]
@jmehnle Could you show the source code and the error message?

**Comment by hauntsaninja** [comment]
Similar to https://github.com/python/mypy/issues/8836
